src/adams_methods.jl

```diff
@@ -0,0 +1,628 @@
+                end
+            end
+            # Keep timeout_VO 0, or decrement
+            timeout_VO= timeout_VO==0? 0 : timeout_VO-1
+        end # end of adaptive order selection
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the
+        # "previous step" position for ϕstar and ϕ
+        # Note: ϕ_n[:] and ϕ_star[2,:] corresponds to ϕ_n and ϕ*_n, and
+        # ϕ_star[1,:] to ϕ*_{n-1}, etc
+        ϕstar_nm1[1:min(max(n-1,1),current_stageorder+1)] =
+                                ϕstar_n[1:min(max(n-1,1),current_stageorder+1)]
+
+        # Update stepsize variables
+        prev_dt = dt
+        dt = next_dt
+
+        # push work variables
+        y_n = y_np1
+        dy_n = dy_np1
+
+        # Increment our step
+        n= n+1
+    end #while loop over steps
+
+    if points == :steps
+        return vcat(t), y
+    else
+        return vcat(tout), yout
+    end
+end
+
+###############################################################################
+#
+# HELPER FUNCTIONS FOR ADAMS VSVO SOLVER
+#
+###############################################################################
+
+###############################################################################
+# Gamma star function from Hairer et al pg 359, table 1.2
+# used in adaptive step selection (See Hairer pg 423)
+###############################################################################
+#precompute factorial_values
+"compute γ*(j) for higher order versions of ode113"
+function gamma_star_constr(index)
+    γ_star = zeros(index)
+    #γ_star[1] = 1
+    for j = 1:index
+        γ_star[j] = Polynomials.polyval((1 / factorial(j))*
+            Polynomials.polyint(Polynomials.poly(diagm(collect(1-(j-1):1)))),1)
+    end
+    return γ_star
+end
+################################################################################
+# Helper functions for calculating coefficients from recurence relationships
+################################################################################
+#Based on Hairer Lemma 5.1 on page 399
+#Calculates c_j,q and g_n,j for n and j =1:stageorder
+"compute c_{j,q} and g_{n,j} coefficients for ode113"
+function g_coefs!(g,c,dt,t,t_np1,n,stageorder)
+    for j = 1:stageorder
+      for q = 1:((stageorder)-(j-1))
+        if j ==1
+          c[j,q]=1/q
+        elseif j==2
+          c[j,q]=1/q/(q+1)
+        else
+          c[j,q] = c[j-1,q] - c[j-1,q+1]*dt/(t_np1-t[n-(j-1)+1])
+        end
+      end
+      g[j]=c[j,1]
+    end
+    return nothing
+end
+
+
+#Based on Hairer equation 5.9. Calculates b, ϕ, ϕstar for
+#n and j  in range_of_indices
+"compute ϕ_{n,j} and ϕ*_{n,j} coefficients for ode113"
+function ϕ_and_ϕstar_coefs!(b, ϕstar_nm1, ϕstar_n, ϕ_n,dy_n,t,t_np1,n,range_of_indices)
```

Comment:
put arguments which are modified in-place first.  Document which are modified in-place.